sparkpost/message_events.py

```diff
@@ -0,0 +1,133 @@
+        else:
+            mesg_filter_str = str(mesg_filter)
+
+        return mesg_filter_str
+
+    def _format_request_params(self, **kwargs):
+        """
+        takes in message event kwargs and returns the request string
+
+        param: str/num/list bounce_classes - Comma-delimited
+            list of bounce classification codes to search
+
+        params dict is a dict of avaliable params.
+            Each param is keyed to a tuple of the
+            (param type, special validator function (if wanted / needed))
+
+            - Listable
+                are params that can be in a list format(commma seperated)
+            - datetime
+                can be a datetime string, or a datetime object
+            - single_val
+                are params that are only single values (ie not lists or dates)
+        """
+
+        # can only be in a certain range
+        def per_page_validator(per_page):
+            per_page = int(per_page)
+            if per_page > 10000 or per_page < 1:
+                raise ValueError("""You entered, Per \
+                    "page must be inbetween 1 and 10,000""")
+            else:
+                return per_page
+
+        params_dict = {
+            # Listable options
+            'bounce_classes': ('listable', None),
+            'campaign_ids': ('listable', None),
+            'events': ('listable', None),
+            'friendly_froms': ('listable', None),
+            'message_ids': ('listable', None),
+            'recipients': ('listable', None),
+            'subaccounts': ('listable', None),
+            'template_ids': ('listable', None),
+            'transmission_ids': ('listable', None),
+            # datetimes
+            'from': ('datetime', None),
+            'to': ('datetime', None),
+            # single vals
+            'page': ('single_val', None),
+            'per_page': ('single_val', per_page_validator),
+            'reason': ('single_val', None),
+            'timezone': ('single_val', None)
+        }
+
+        request_list = []
+        for param, filter_val in kwargs.iteritems():
+            param_type, special_validator = params_dict[param]
+
+            if special_validator is not None:
+                filter_val = special_validator(filter_val)
+
+            if param_type == "listable":
+                filter_val = self._format_listable_param(filter_val)
+            elif param_type == "datetime":
+                filter_val = self._format_datetime_param(filter_val)
+            else:
+                filter_val = str(filter_val)
+
+            request_list.append("{param}={filter_val}".format(
+                param=param,
+                filter_val=filter_val
+            ))
+
+        request_str = "&".join(request_list)
+        return request_str
+
+    def _fetch_get(self, request_str):
+        uri = "%s?%s" % (self.uri, request_str)
+        results = self.request('GET', uri, return_all_keys=True)
```

Comment:
I'm all for including the `return_all_keys` flag. Can you talk more about the use case behind setting this to `True` here? My concern is having a totally different payload coming through for this API. I wonder if we should make this an options that passes through as an arg on the `get` method.